### PR TITLE
Support Django 4.1, 4.2 and Python 3.11

### LINKED ISSUE #486: Support Django 4.2
# Feature Request

Add Django 4.2 to the automated CI testing suite.

# Issue

When attempting to add support for Django 4.2, the following error appears within CI:

```
ERROR Error sending data
  Traceback (most recent call last):
    File "/home/runner/work/django-dbbackup/django-dbbackup/.tox/py310-django42/lib/python3.10/site-packages/gnupg.py", line 155, in _copy_data
      outstream.write(data)
  TypeError: a bytes-like object is required, not 'str'
  ....ERROR Exception: Foo
    File "/home/runner/work/django-dbbackup/django-dbbackup/dbbackup/utils.py", line 120, in wrapper
      func(*args, **kwargs)
    File "/home/runner/work/django-dbbackup/django-dbbackup/dbbackup/tests/test_utils.py", line 81, in func
      raise Exception("Foo")
  
  .ERROR Exception: Foo
    File "/home/runner/work/django-dbbackup/django-dbbackup/dbbackup/utils.py", line 120, in wrapper
      func(*args, **kwargs)
    File "/home/runner/work/django-dbbackup/django-dbbackup/dbbackup/tests/test_utils.py", line 91, in func
      raise Exception("Foo")
  
  ..ERROR Error sending data
  Traceback (most recent call last):
    File "/home/runner/work/django-dbbackup/django-dbbackup/.tox/py310-django42/lib/python3.10/site-packages/gnupg.py", line 155, in _copy_data
      outstream.write(data)
  TypeError: a bytes-like object is required, not 'str'
```

https://github.com/jazzband/django-dbbackup/blob/790ec8bd297ef24cddfd2618d18d243abd15437e/dbbackup/tests/utils.py#L120-L122

https://github.com/jazzband/django-dbbackup/blob/790ec8bd297ef24cddfd2618d18d243abd15437e/dbbackup/tests/test_utils.py#L87-L100

### `python-gnupg`

https://github.com/jazzband/django-dbbackup/blob/790ec8bd297ef24cddfd2618d18d243abd15437e/requirements/tests.txt#L9

https://github.com/vsajip/python-gnupg/blob/6081e55ccb513a478a64eaa48e993cb40853851e/gnupg.py#L138

### PR DESCRIPTION
**Pull request by johnthagen**
# CI

Closes #486 

## Description

- Add support for Django 4.1 & 4.2
- Add support for Python 3.11
- ~~Drop support for Python 3.7 ([end of life in 1 month](https://endoflife.date/python))~~ Had to back out due to https://github.com/jazzband/help/issues/336
- Fix compatibility with tox 4
- Update `isort` precommit version to fix CI error

### CODE REVIEW

**Comment by johnthagen** [comment]
Error under Django 4.2 seems to be:

```
ERROR Error sending data
  Traceback (most recent call last):
    File "/home/runner/work/django-dbbackup/django-dbbackup/.tox/py310-django42/lib/python3.10/site-packages/gnupg.py", line 155, in _copy_data
      outstream.write(data)
  TypeError: a bytes-like object is required, not 'str'
  ....ERROR Exception: Foo
    File "/home/runner/work/django-dbbackup/django-dbbackup/dbbackup/utils.py", line 120, in wrapper
      func(*args, **kwargs)
    File "/home/runner/work/django-dbbackup/django-dbbackup/dbbackup/tests/test_utils.py", line 81, in func
      raise Exception("Foo")
  
  .ERROR Exception: Foo
    File "/home/runner/work/django-dbbackup/django-dbbackup/dbbackup/utils.py", line 120, in wrapper
      func(*args, **kwargs)
    File "/home/runner/work/django-dbbackup/django-dbbackup/dbbackup/tests/test_utils.py", line 91, in func
      raise Exception("Foo")
  
  ..ERROR Error sending data
  Traceback (most recent call last):
    File "/home/runner/work/django-dbbackup/django-dbbackup/.tox/py310-django42/lib/python3.10/site-packages/gnupg.py", line 155, in _copy_data
      outstream.write(data)
  TypeError: a bytes-like object is required, not 'str'
```

**Comment by johnthagen** [comment]
@Archmonger If you could review that would be great. The project settings will need to be updated so that Python 3.7 is no longer required, and Python 3.11 is set to be required.

**Comment by Archmonger** [comment]
Will review soon, but unfortunately I cannot change the GitHub test requirements.

We'll have to wait until Janis transfers us out of Jazzband.

File: requirements/tests.txt [hunk 1]
```diff
@@ -9,5 +9,5 @@ python-dotenv
 python-gnupg>=0.5.0
 pytz
 testfixtures
-tox
+tox>=4.0.0
```
Comment (Archmonger) [inline requirements/tests.txt:12]: We should create an issue to replace `tox` with `nox`.

**Review by Archmonger — APPROVED** [review]
LGTM. Unfortunately can't merge until the jazzband situation gets addressed.

File: requirements/tests.txt (hunk 1 above)
Comment (johnthagen) [reply requirements/tests.txt:12]: - #487

**Pull request merged**